[2/N] Remove unittest.skip across all of torch.distributed.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #61887

1) Introduced a `sandcastle_skip_if` decorator that ensures these
tests just get passed on sandcastle.
2) Fixed all test files under `test/distributed` to not use `unittest.skip`

Overall goal is to avoid using skips since sandcastle tags these tests as
continuously skipping.

Differential Revision: [D29784152](https://our.internmc.facebook.com/intern/diff/D29784152/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29784152/)!

pritamdamania committed:
[2/N] Remove unittest.skip across all of torch.distributed.

1) Introduced a `sandcastle_skip_if` decorator that ensures these
tests just get passed on sandcastle.
2) Fixed all test files under `test/distributed` to not use `unittest.skip`

Overall goal is to avoid using skips since sandcastle tags these tests as
continuously skipping.

Differential Revision: [D29784152](https://our.internmc.facebook.com/intern/diff/D29784152/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29784152/)!

[ghstack-poisoned]

rohan-varma commented on test/distributed/test_pg_wrapper.py:
```diff
@@ -212,177 +212,175 @@ def _test_collective_shape_mismatch(self, wrapper_pg, use_cuda=False):
+        @requires_nccl()
+        @skip_if_lt_x_gpu(2)
+        @with_dist_debug_levels(levels=["OFF"])
+        @sandcastle_skip_if(torch.cuda.device_count() < 2, "NCCL test requires 2+ GPUs")
```
Not sure I understand why we need both `sandcastle_skip_if` and then skip_if_lt_x_gpu_2 as well here? Shouldn't we only need the sandcastle one?

rohan-varma commented on test/distributed/test_pg_wrapper.py:
```diff
@@ -212,177 +212,175 @@ def _test_collective_shape_mismatch(self, wrapper_pg, use_cuda=False):
+            pg = self._create_wrapper_pg(with_new_group=True)
+            self._test_collective_shape_mismatch(pg, use_cuda=True)
+
+        @skip_if_lt_x_gpu(4)
```
Should these be sandcastle skip if?

pritamdamania committed:
Update on "[2/N] Remove unittest.skip across all of torch.distributed."


1) Introduced a `sandcastle_skip_if` decorator that ensures these
tests just get passed on sandcastle.
2) Fixed all test files under `test/distributed` to not use `unittest.skip`

Overall goal is to avoid using skips since sandcastle tags these tests as
continuously skipping.

Differential Revision: [D29784152](https://our.internmc.facebook.com/intern/diff/D29784152/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29784152/)!

[ghstack-poisoned]

pritamdamania committed:
Update on "[2/N] Remove unittest.skip across all of torch.distributed."


1) Introduced a `sandcastle_skip_if` decorator that ensures these
tests just get passed on sandcastle.
2) Fixed all test files under `test/distributed` to not use `unittest.skip`

Overall goal is to avoid using skips since sandcastle tags these tests as
continuously skipping.

Differential Revision: [D29784152](https://our.internmc.facebook.com/intern/diff/D29784152/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29784152/)!

[ghstack-poisoned]

pritamdamania committed:
Update on "[2/N] Remove unittest.skip across all of torch.distributed."


1) Introduced a `sandcastle_skip_if` decorator that ensures these
tests just get passed on sandcastle.
2) Fixed all test files under `test/distributed` to not use `unittest.skip`

Overall goal is to avoid using skips since sandcastle tags these tests as
continuously skipping.

Differential Revision: [D29784152](https://our.internmc.facebook.com/intern/diff/D29784152/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29784152/)!

[ghstack-poisoned]

pritamdamania committed:
Update on "[2/N] Remove unittest.skip across all of torch.distributed."


1) Introduced a `sandcastle_skip_if` decorator that ensures these
tests just get passed on sandcastle.
2) Fixed all test files under `test/distributed` to not use `unittest.skip`

Overall goal is to avoid using skips since sandcastle tags these tests as
continuously skipping.

Differential Revision: [D29784152](https://our.internmc.facebook.com/intern/diff/D29784152/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29784152/)!

[ghstack-poisoned]

pritamdamania committed:
Update on "[2/N] Remove unittest.skip across all of torch.distributed."


1) Introduced a `sandcastle_skip_if` decorator that ensures these
tests just get passed on sandcastle.
2) Fixed all test files under `test/distributed` to not use `unittest.skip`

Overall goal is to avoid using skips since sandcastle tags these tests as
continuously skipping.

Differential Revision: [D29784152](https://our.internmc.facebook.com/intern/diff/D29784152/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29784152/)!

[ghstack-poisoned]

pritamdamania committed:
Update on "[2/N] Remove unittest.skip across all of torch.distributed."


1) Introduced a `sandcastle_skip_if` decorator that ensures these
tests just get passed on sandcastle.
2) Fixed all test files under `test/distributed` to not use `unittest.skip`

Overall goal is to avoid using skips since sandcastle tags these tests as
continuously skipping.

Differential Revision: [D29784152](https://our.internmc.facebook.com/intern/diff/D29784152/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29784152/)!

[ghstack-poisoned]

pritamdamania committed:
Update on "[2/N] Remove unittest.skip across all of torch.distributed."


1) Introduced a `sandcastle_skip_if` decorator that ensures these
tests just get passed on sandcastle.
2) Fixed all test files under `test/distributed` to not use `unittest.skip`

Overall goal is to avoid using skips since sandcastle tags these tests as
continuously skipping.

Differential Revision: [D29784152](https://our.internmc.facebook.com/intern/diff/D29784152/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29784152/)!

[ghstack-poisoned]

pritamdamania committed:
Update on "[2/N] Remove unittest.skip across all of torch.distributed."


1) Introduced a `sandcastle_skip_if` decorator that ensures these
tests just get passed on sandcastle.
2) Fixed all test files under `test/distributed` to not use `unittest.skip`

Overall goal is to avoid using skips since sandcastle tags these tests as
continuously skipping.

Differential Revision: [D29784152](https://our.internmc.facebook.com/intern/diff/D29784152/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29784152/)!

[ghstack-poisoned]

pritamdamania committed:
Update on "[2/N] Remove unittest.skip across all of torch.distributed."


1) Introduced a `sandcastle_skip_if` decorator that ensures these
tests just get passed on sandcastle.
2) Fixed all test files under `test/distributed` to not use `unittest.skip`

Overall goal is to avoid using skips since sandcastle tags these tests as
continuously skipping.

Differential Revision: [D29784152](https://our.internmc.facebook.com/intern/diff/D29784152/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29784152/)!

[ghstack-poisoned]

pritamdamania committed:
Update on "[2/N] Remove unittest.skip across all of torch.distributed."


1) Introduced a `sandcastle_skip_if` decorator that ensures these
tests just get passed on sandcastle.
2) Fixed all test files under `test/distributed` to not use `unittest.skip`

Overall goal is to avoid using skips since sandcastle tags these tests as
continuously skipping.

Differential Revision: [D29784152](https://our.internmc.facebook.com/intern/diff/D29784152/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29784152/)!

[ghstack-poisoned]
